File: backend/src/attendance/dto/finalize-session.dto.ts
import { IsString, IsNotEmpty, IsArray } from 'class-validator';

export class FinalizeSessionDto {
    @IsString()
    @IsNotEmpty()
    sessionId: string;

    @IsString()
    @IsNotEmpty()
    teacherId: string;

    @IsString()
    @IsNotEmpty()
    courseId: string;

    @IsArray()
    @IsNotEmpty()
    recordsJson: any[];

    @IsString()
    @IsNotEmpty()
    idempotencyKey: string;
}

File: android/app/src/main/kotlin/com/smartpresence/idukay/data/remote/dto/FinalizeSessionDto.kt
package com.smartpresence.idukay.data.remote.dto

data class FinalizeSessionRequest(
    val sessionId: String,
    val teacherId: String,
    val courseId: String,
    val recordsJson: List<AttendanceRecordDto>,
    val idempotencyKey: String
)

data class FinalizeSessionResponse(
    val success: Boolean,
    val status: String, // "applied", "already_applied", "rejected"
    val finalizationId: String
)
